woodchipper/context.py

```diff
@@ -63,12 +67,15 @@ class LoggingContext:
     ```
     """
 
-    def __init__(self, injected_context: LoggingContextType):
+    def __init__(self, injected_context: LoggingContextType, prefix=MISSING_PREFIX):
         self.injected_context = injected_context
+        self.prefix = os.getenv("WOODCHIPPER_KEY_PREFIX") if prefix is MISSING_PREFIX else prefix
         self._token = None
 
     def __enter__(self):
-        self._token = logging_ctx.update(self.injected_context)
+        self._token = logging_ctx.update(
+            {(f"{self.prefix}.{k}" if self.prefix else k): v for k, v in self.injected_context.items()}
+        )
```

Comment:
Is it okay if the log keys are `'None.foo'`? We'll have that if prefix is missing and envvar is missing as well?

That may be an issue for key-collision reasons now that I think of it. Both new and old values would have `None.whatver`, so if they shared a key, the new would overwrite the old.

Reply:
Resolved in conversation :)